synapse/handlers/event_auth.py

```diff
@@ -160,6 +160,27 @@ async def check_host_in_room(self, room_id: str, host: str) -> bool:
         with Measure(self._clock, "check_host_in_room"):
             return await self._store.is_host_joined(room_id, host)
 
+    async def assert_host_in_room(
+        self, room_id: str, host: str, allow_partial_state_rooms: bool = False
+    ) -> None:
+        """
+        Asserts that the host is in the room, or raises an AuthError.
+
+        If the room is partial-stated, we raise an AuthError with the
+        UNABLE_DUE_TO_PARTIAL_STATE flag, unless `allow_partial_state_rooms` is true.
```

Comment:
Suppose `allow_partial_state_rooms=True`. If this function returns True, am I definitely in the room? If it retusn False, am I definitely _not_ in the room right now?

I'm wondering if this needs some kind of warning. e.g. "if `allow_partial_state_rooms` is true, you need to handle the fact that this function may return an incorrect result." But I'm lack context here, I might just be making noise...

Reply:
that sounds like a good point.